File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0 ;
          vector<int> ans;
        int right = numbers.size() - 1;
        while(left < right){
            if (numbers[left] + numbers[right] == target){
                ans.push_back(left + 1);
                ans.push_back(right + 1);
                return ans;
            }
            if (numbers[left] + numbers[right] < target)
                left++;
            else
                right --;
        }
        return ans;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        dp = defaultdict(int)
        inex = defaultdict(int)
        for i in range(len(numbers)):
            if dp[target - numbers[i]] > 0:
                return [inex[target - numbers[i]], i + 1]
            inex[numbers[i]] = i + 1
            dp[numbers[i]]+=1
        